src/serial.rs

```diff
@@ -111,6 +116,97 @@ cfg_if! {
     }
 }
 
+/// Types for configuring a serial interface.
+pub mod config {
+    use crate::time::rate::{Baud, Extensions};
+
+    // Reexport stop bit enum from PAC. In case there is a breaking change,
+    // provide a compatible enum from this HAL.
+    pub use crate::pac::usart1::cr2::STOP_A as StopBits;
+
+    /// Parity generation and checking. If odd or even parity is selected, the
+    /// underlying USART will be configured to send/receive the parity bit in
+    /// addtion to the data bits.
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    #[derive(Clone, Copy, PartialEq)]
+    pub enum Parity {
+        /// No parity bit will be added/checked.
+        ParityNone,
+        /// The MSB transmitted/received will be generated/checked to have a
+        /// even number of bits set.
+        ParityEven,
+        /// The MSB transmitted/received will be generated/checked to have a
+        /// odd number of bits set.
+        ParityOdd,
+    }
+
+    /// Configuration struct for [`Serial`](super::Serial) providing all
+    /// communication-related / parameters.
+    #[derive(Clone, Copy, PartialEq)]
+    pub struct Config {
+        /// Serial interface baud rate
+        pub baudrate: Baud,
+        /// Whether and how to generate/check a parity bit
+        pub parity: Parity,
+        /// The number of stop bits to follow the last data bit or the parity
+        /// bit
+        pub stopbits: StopBits,
+    }
+
+    impl Config {
```

Comment:
Add a `new` method, wrapping `default`

```suggestion
    impl Config {
        fn new() -> Self {
            Config::default()
        }
```

Reply:
Using `new` for a default initialization feels a bit strange to me. What about a `new` with the most prominent parameters (baudrate, parity, stop bits - and yes, we have not more at the moment) and `default` for initializing them too with defaults.

Future additional parameters could get default values in `new` (and will with the current state of the code).

What do you think of this approach?